domain/machine/state/machine_cloud_instance.go

```diff
@@ -148,3 +150,52 @@ WHERE machine_uuid=$instanceTag.machine_uuid
 		return nil
 	})
 }
+
+// InstanceId returns the cloud specific instance id for this machine.
+// If the machine is not provisioned, it returns a NotProvisionedError.
+func (st *State) InstanceId(ctx context.Context, machineId coremachine.ID) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	machineIDParam := sqlair.M{"machine_id": machineId}
+	query := `
+SELECT instance_id AS &instanceID.*
+FROM machine AS m
+         JOIN machine_cloud_instance AS mci ON m.uuid = mci.machine_uuid
+WHERE m.machine_id = $M.machine_id;
+`
+	queryStmt, err := st.Prepare(query, sqlair.M{}, instanceID{})
```

Comment:
There's no need to create an empty map here when you have the one created above - just use that.

Furthermore, we agreed with @tlm to call what was heretofore "machine ID" "machine name". We've already gone this way with units.

So create a type with just the machine name, and use that struct to parameterise all queries using "ID".

We can do the other renaming (such as in core) later.